internal/k8s/controllers/layer2_status_controller.go

```diff
@@ -99,40 +125,89 @@ func (r *Layer2StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	level.Debug(r.Logger).Log("controller", "Layer2StatusReconciler", "updated state", dumpResource(state))
-
 	return ctrl.Result{}, nil
 }
 
 func (r *Layer2StatusReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	p := predicate.NewPredicateFuncs(func(object client.Object) bool {
 		if s, ok := object.(*v1beta1.ServiceL2Status); ok {
-			return strings.HasSuffix(s.Name, r.NodeName)
+			// only objects with complete labels that can illustrate which service it is related to
+			// can trigger the reconciler
+			label := s.GetLabels()
+			if label == nil {
+				level.Error(r.Logger).Log("controller", "Layer2StatusReconciler", "missing meta", "object", object)
+				return false
+			}
+			if _, ok = label[LabelServiceName]; !ok {
+				level.Error(r.Logger).Log("controller", "Layer2StatusReconciler", "missing meta", "object", object)
+				return false
+			}
+			if _, ok = label[LabelServiceNamespace]; !ok {
+				level.Error(r.Logger).Log("controller", "Layer2StatusReconciler", "missing meta", "object", object)
+				return false
+			}
+			var node string
+			if node, ok = label[LabelAnnounceNode]; !ok {
+				level.Error(r.Logger).Log("controller", "Layer2StatusReconciler", "missing meta", "object", object)
+				return false
+			}
+			// only trigger the reconciler if the service is announced by this node
+			if node != r.NodeName {
```

Comment:
There is no
```
level.Error(r.Logger).Log("controller", "Layer2StatusReconciler", "missing meta", "object", object)
```
before thel last `return false`, while there is in all previous `return false`

Reply:
That's because this is the happy path (the node can or can not be the current node) while the others are exceptional conditions that are worth labeling.